### DISCUSSION TITLE
Configure cleanup policy for release history

### DISCUSSION THREAD

**Discussion by klboke** [apolloconfig/apollo#4667] [General]
###  Issues
`releasehistory`、`commit`、`release` These tables will keep growing as the application configuration is released, until the query bottleneck is reached and the UI cannot be opened.

### Solutions
The release history, in general, will only focus on recent releases, and historical releases basically have no value of existence, so we can specify some policies to keep this data in a controlled range, such as :
- Keep the last 100 posting history data
- Retain data for the last 30 days of time

Of course, the strategies are configurable drivers, and the deletion is also a physical deletion, because the logical deletion can not reach the effect of cleaning
cc @apolloconfig/pmc

**Comment by Anilople** [comment]
Configdb really need a cleanup strategy.

More information:
* https://github.com/apolloconfig/apollo/issues/3208

**Comment by nobodyiam** [comment]
It's reasonable to config a policy to save the database space.
However, there might be some more aspects we need to consider, e.g.
1. Whether to trigger the clean-up process automatically or manually?
2. Is it necessary to config the clean-up policy at the app level as well as the global level?
3. Is the clean-up policy different among environments?

> These tables will keep growing as the application configuration is released, until the query bottleneck is reached and the UI cannot be opened.

As the database has indices, I suppose `until the query bottleneck is reached and the UI cannot be opened` will not happen?

**Comment by klboke** [comment]
> As the database has indices, I suppose until the query bottleneck is reached and the UI cannot be opened will not happen?

Some of our projects have the configuration in yaml format, and then the configuration is very large, so it exacerbates the problem

**Comment by klboke** [comment]
- PR：https://github.com/apolloconfig/apollo/pull/4813
